File: app/src/main/java/ua/com/foxminded/locationtrackera/di/AppComponent.java
package ua.com.foxminded.locationtrackera.di;

import javax.inject.Singleton;

import dagger.Component;

import ua.com.foxminded.locationtrackera.background.LocationService;
import ua.com.foxminded.locationtrackera.background.jobs.LocationsUploader;
import ua.com.foxminded.locationtrackera.ui.MapsActivity;
import ua.com.foxminded.locationtrackera.ui.TrackerActivity;
import ua.com.foxminded.locationtrackera.ui.auth.AuthViewModelFactory;

@Component(modules = {AppModule.class, ServiceModule.class, DataModule.class})
@Singleton
public interface AppComponent {
    void inject(AuthViewModelFactory factory);
    void inject(LocationService service);
    void inject(LocationsUploader worker);
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/di/AppComponent.kt
package ua.com.foxminded.locationtrackera.di

import dagger.Component
import ua.com.foxminded.locationtrackera.background.LocationService
import ua.com.foxminded.locationtrackera.background.jobs.LocationsUploader
import ua.com.foxminded.locationtrackera.ui.auth.AuthViewModelFactory
import javax.inject.Singleton

@Component(modules = [AppModule::class, ServiceModule::class, DataModule::class])
@Singleton
interface AppComponent {
    fun inject(factory: AuthViewModelFactory)
    fun inject(service: LocationService)
    fun inject(worker: LocationsUploader)
}
